### PR TITLE
Add results-directory switch to dotnet test

### PR DESCRIPTION
**Pull request by smadala**
- Fixes: https://github.com/Microsoft/vstest/issues/299
- We took this earlier [here](https://github.com/dotnet/cli/pull/5391) and reverted  [here](https://github.com/dotnet/cli/pull/5541) as it was not met the RTW bar.

@livarcocc @codito @piotrpMSFT

### CODE REVIEW

File: test/dotnet-test.Tests/GivenDotnetTestBuildsAndRunsTestfromCsproj.cs [hunk 1]
```diff
@@ -109,7 +109,7 @@ public void TestWillCreateTrxLoggerInTheSpecifiedResultsDirectoryBySwitch()
             // Call test with trx logger enabled and results directory explicitly specified.
             CommandResult result = new DotnetTestCommand()
                                        .WithWorkingDirectory(testProjectDirectory)
-                                       .ExecuteWithCapturedOutput("--logger:trx -r \"" + trxLoggerDirectory + "\"");
+                                       .ExecuteWithCapturedOutput("--logger trx -r \"" + trxLoggerDirectory + "\"");
```
Comment (livarcocc) [inline test/dotnet-test.Tests/GivenDotnetTestBuildsAndRunsTestfromCsproj.cs]: Why change this? This format should still be supported.

File: src/dotnet/commands/dotnet-test/LocalizableStrings.cs [hunk 2]
```diff
@@ -60,5 +60,10 @@ internal class LocalizableStrings
 
         public const string RunSettingsArgsHelpText = @"Any extra command-line runsettings arguments that should be passed to vstest. See 'dotnet vstest --help' for available options.
                                         Example: -- RunConfiguration.ResultsDirectory=""C:\users\user\desktop\Results Directory"" MSTest.DeploymentEnabled=false";
+
+        public const string CmdResultsDirectoryDescription = @"The test results directory will be created in the specified path if it does not exist.
```
Comment (livarcocc) [inline src/dotnet/commands/dotnet-test/LocalizableStrings.cs]: This seems like a weird description for this parameter. Should it say something like: "The directory where tests results will be placed. If the specified directory does not exist, one will be created for you."

@mairaw

File: test/dotnet-test.Tests/GivenDotnetTestBuildsAndRunsTestfromCsproj.cs (hunk 1 above)
Comment (smadala) [inline test/dotnet-test.Tests/GivenDotnetTestBuildsAndRunsTestfromCsproj.cs]: This was a conflict from https://github.com/dotnet/cli/commit/cb52224272cda0e148dbd0f88332f21dcc508750#diff-56c1a6ebe6851ff03faea7b182cb783e.

File: src/dotnet/commands/dotnet-test/LocalizableStrings.cs (hunk 2 above)
Comment (smadala) [inline src/dotnet/commands/dotnet-test/LocalizableStrings.cs]: Update to "The directory where tests results will be placed. Specified directory will be created if it does not exist."

**Comment by smadala** [comment]
@dotnet-bot Test Windows_NT x64 Release Build, Ubuntu x64 Release Build, RHEL7.2 x64 Release Build

**Comment by smadala** [comment]
@dotnet-bot Test Ubuntu x64 Release Build

**Comment by smadala** [comment]
@dotnet-bot Test RHEL7.2 x64 Release Build

**Comment by smadala** [comment]
@livarcocc Can you please review again?

**Pull request merged**